overview/helloWorld/4-iexecApplication.md

```diff
@@ -0,0 +1,251 @@
+```
+
+If you want, you can integrate protected data into your iDapp to process private
+data. To do that, you need to start this tutorial again and answer `y` to the
+question `? Would you like to access protected data inside your iDapp?`.
+
+<!-- ## 🧩 Let's Create Protected Data
+
+To create protected data, you need to connect your wallet and create the
+protected data using the provided buttons. -->
+
+<script setup>
+import { ref } from 'vue';
+import { IExecDataProtectorCore } from '@iexec/dataprotector';
+import MetamaskButton from '../../components/MetamaskButton.vue';
+
+const web3Provider = ref(null);
+const isWalletConnected = ref(false);
+const protectedData = ref(null);
+const authorizedApp = ref('');
+const contentToProtect = ref('');
+const isLoadingProtect = ref(false);
+const isLoadingGrant = ref(false);
+const protectError = ref(null);
+const grantError = ref(null);
+
+const onWalletConnected = (provider) => {
+  web3Provider.value = provider;
+  isWalletConnected.value = true;
+};
+
+// const protectData = async () => {
+//   try {
+//     if (!web3Provider.value) throw new Error('Wallet not connected');
+//     if (!contentToProtect.value) throw new Error('Content is empty');
+//     isLoadingProtect.value = true;
+//     protectError.value = null;
+//     const dataProtectorCore = new IExecDataProtectorCore(web3Provider.value);
+//     protectedData.value = await dataProtectorCore.protectData({
+//       data: {
+//         content: contentToProtect.value,
+//       },
+//     });
+//   } catch (error) {
+//     protectError.value = error.message;
+//     console.error('Error protecting data:', error);
+//   } finally {
+//     isLoadingProtect.value = false;
+//   }
+// };
+
+const grantAccess = async () => {
+  try {
+    if (!web3Provider.value || !protectedData.value) throw new Error('Missing data');
+    isLoadingGrant.value = true;
+    grantError.value = null;
+    const dataProtectorCore = new IExecDataProtectorCore(web3Provider.value);
+    const grantedAccess = await dataProtectorCore.grantAccess({
+      protectedData: protectedData.value.address,
+      authorizedApp: authorizedApp.value,
+      authorizedUser: '0x0000000000000000000000000000000000000000',
+    });
+    console.log('Access granted:', grantedAccess);
+  } catch (error) {
+    grantError.value = error.message;
+    console.error('Error granting access:', error);
+  } finally {
+    isLoadingGrant.value = false;
+  }
+};
+</script>
+
+<!-- <div class="form-container">
+  <input v-model="contentToProtect" placeholder="Enter content to protect" />
+  <button @click="protectData" :disabled="!isWalletConnected || isLoadingProtect">
+    {{ isLoadingProtect ? 'Processing...' : 'Protect Data' }}
+  </button>
+  <div v-if="protectError" class="error">{{ protectError }}</div>
+</div>
+
+<div v-if="protectedData">
+  <h2>Protected Data Address:</h2>
+  <p>{{ protectedData.address }}</p>
+</div> -->
+
+## 🧩 Grant Access to Your iDapp
```

Comment:
I would create another section to explain all of that.
"5-grantAccess", something like that.

Idea:

> "Great, now that we have created a protected data in `3-protectAndManageData` and an iExec app in `4-iexecApplication`, we can combine both of these and grant an access for someone to use your protected data with this specific iDapp."